command/agent/config/config.go

```diff
@@ -40,6 +40,7 @@ type Vault struct {
 	ClientCert       string      `hcl:"client_cert"`
 	ClientKey        string      `hcl:"client_key"`
 	TLSServerName    string      `hcl:"tls_server_name"`
+	Retry            *Retry      `hcl:"retry"`
```

Comment:
The retry behavior in here is specifically for the template server (at least for now) so I'm not entirely convinced that it should live in the `vault` stanza since this doesn't control retry logic when agent communicates with the server for auto-auth and/or caching.

@HridoyRoy thoughts on making this be a top-level `template_retry` stanza instead?

Reply:
We did think about that, but the existing retry context is under `ctv.Vault.Retry = &ctconfig.RetryConfig{Attempts: &ts.testingLimitRetry}` and consul-template puts it under the vault section so we kept the same structure location.

As for the auto-auth retry that is a feature we would like to add as well as our use case of vault needs to be fault tolerant of network disconnects both for getting new tokens and template rendering, this MR was the first step.

Reply:
Hm, I do agree that "Retry" in the Vault section shouldn't be a configuration parameter that is limited to the consul template use-case. Let me dig into this a bit -- I'll circle back soon.

Reply:
Hi @ibspoof @RXC001 , sorry for the delay in circling back! 

We'd like to move the Retry stanza into as a top level config named `template_retry`. `ctv` is the consul template, so `ctv.Vault.Retry ` can be set to values that exist in a structurally different location in the Vault config. Since this retry functionality is currently limited in scope to the consul-template, we'd want to make that clear with the name and location of the stanza, instead of keeping structural similarity between the consul-template config and the vault config. 

Thanks so much!

Reply:
Makes sense we had the same discussion in our team. Will get that changed and pushed.